core/src/test/scala/org/apache/spark/storage/BlockManagerDecommissionIntegrationSuite.scala

```diff
@@ -213,10 +213,7 @@ class BlockManagerDecommissionIntegrationSuite extends SparkFunSuite with LocalS
       assert(execIdToBlocksMapping.values.flatMap(_.keys).count(_.isRDD) === numParts)
     }
 
-    // Make the executor we decommissioned exit
-    sched.client.killExecutors(List(execToDecommission))
```

Comment:
I think we are missing some tests to ensure that the executors do actually exit. Particularly in light of @attilapiros recent comment :-P

Reply:
So that’s the point of taking out the kill executor. If you look a line bellow we wait on the semaphore to make sure the executor has exited.